carmm/run/aims_calculator.py

```diff
@@ -227,47 +227,32 @@ def get_k_grid(model, sampling_density, verbose=False, simple_reciprocal_space_p
         return None
 
     # These are lattice vectors
-    x_v = model.get_cell()[0]
-    y_v = model.get_cell()[1]
-    z_v = model.get_cell()[2]
+    lattice_v = np.array(model.get_cell())
     # These are lattice parameters
-    x = np.linalg.norm(x_v)
-    y = np.linalg.norm(y_v)
-    z = np.linalg.norm(z_v)
+    lattice_param = np.array([np.linalg.norm(v) for v in lattice_v])
 
     if simple_reciprocal_space_parameters:
         # Simplified reciprocal lattice parameters
-        l_v_x = 2 * math.pi / x
-        l_v_y = 2 * math.pi / y
-        l_v_z = 2 * math.pi / z
+        reciprocal_param = 2 * math.pi / lattice_param
     else:
         # volume of the cell
-        volume = np.dot(x_v, np.cross(y_v, z_v))
+        volume = np.dot(lattice_v[0], np.cross(lattice_v[1], lattice_v[2]))
         # These are reciprocal lattice vectors
-        v_x = np.cross(y_v, z_v) * 2 * math.pi / volume
-        v_y = np.cross(z_v, x_v) * 2 * math.pi / volume
-        v_z = np.cross(x_v, y_v) * 2 * math.pi / volume
+        reciprocal_v = [np.cross(lattice_v[(i + 1) % 3], lattice_v[(i + 2) % 3]) * 2 * math.pi / volume
+                        for i in range(len(lattice_v))]
         # These are reciprocal lattice parameters
-        l_v_x = np.linalg.norm(v_x)
-        l_v_y = np.linalg.norm(v_y)
-        l_v_z = np.linalg.norm(v_z)
+        reciprocal_param = np.array([np.linalg.norm(r_v) for r_v in reciprocal_v])
 
     k_grid_density = 1 / (sampling_density * 2 * math.pi)
-
-    k_x = math.ceil(k_grid_density * l_v_x)
-    k_y = math.ceil(k_grid_density * l_v_y)
-    k_z = math.ceil(k_grid_density * l_v_z)
-
+    k_grid = k_grid_density * reciprocal_param
+    # Convert k_grid to integer
+    k_grid = np.array([math.ceil(k) for k in k_grid])
```

Comment:
Do you need to typecast this as an integer? As ceil returns a rounded float?

Reply:
This seems to be a problem, but the previous and current version somehow both works fine without typecast k_grid to integer.